### ISSUE TITLE
Click to go to original location functionality

### ISSUE THREAD

**Issue by cyrus-** [hazelgrove/hazel#791]
- holes
- cast errors (using blame)
- function values
- pieces of syntax in indet output (using unique IDs passed down to dhexp?)

**Comment by YC-Qiu** [comment]
Does the 'cast errors" refer to errors of type conversion? I'm also not sure about what kind of performance we want function values to achieve. And can you also specify the last one 'pieces of syntax in indet output (using unique IDs passed down to dhexp?)'?

**Comment by cyrus-** [comment]
Yes, cast errors refer to failed casts (see POPL 2019).

For function values, see #783 -- we have a PR that is adding function names, but there can be many functions with the same name so we could also add unique IDs for that name's binding so that when you click on the function name, you get taken to it.

Indet output is indeterminate output -- again, see POPL 2019. Would be nice to be able to click on any piece of syntax in the output and get its corresponding source location. Might be wise to first finish #857 for that though.

**Comment by YC-Qiu** [comment]
It seems like this issue have dependencies on #783 and #857. Should I find another issue to work on until they finish, or participate in these issues?

**Comment by cyrus-** [comment]
Yeah you may want to wait for #857 to land. You could see if any help is
needed there, or you can pick another smaller project?
